src/mbgl/style/conversion/function.cpp

```diff
@@ -78,7 +78,7 @@ optional<PropertyExpression<T>> convertFunctionToExpression(const Convertible& v
         return nullopt;
     }
 
-    optional<T> defaultValue;
+    optional<T> defaultValue{};
```

Comment:
is this just a cosmetic change or is that actually changing the behavior?

Reply:
that's fixing a compiler warning:
```
cc1plus: error: ‘defaultValue’ may be used uninitialized in this function [-Werror=maybe-uninitialized]
../../../src/mbgl/style/conversion/function.cpp:81:17: note: ‘defaultValue’ was declared here
     optional<T> defaultValue;
```

Reply:
Ah, it's initializing `defaultValue` with the default constructor of `T`.

Reply:
as from https://en.cppreference.com/w/cpp/experimental/optional/optional
`1) Constructs the object that does not contain a value.`

Reply:
though we can be more explicit and use
`optional<T> defaultValue{nullopt}`

Reply:
How is that fixing the compiler warning then?

Reply:
not sure :) this looks like a false warning as putting `{}` just calls the optional's default constructor, i.e. does not change anything